src/App.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import {
+  addTodoToServer,
+  deleteTodoFromServer,
+  editTodo,
+  getTodos,
+} from './api/api';
+import { Todo } from './types/Todo';
+
+const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodo, setVisibleTodo] = useState<Todo[]>([]);
+  const [searchValue, setSearchValue] = useState<Todo['title']>('');
+  const [filterValue, setFilterValue] = useState<string>('all');
+
+  const fetchTodos = async () => {
+    try {
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+      setVisibleTodo(todosFromServer);
+    } catch {
+      Promise.reject(new Error('error'));
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+  }, []);
+
+  useEffect(() => {
+    switch (filterValue) {
+      case 'active':
+        setVisibleTodo(todos.filter(todo => !todo.completed));
+        break;
+      case 'completed':
+        setVisibleTodo(todos.filter(todo => todo.completed));
+        break;
+      default:
+        setVisibleTodo(todos);
+        break;
+    }
+  }, [filterValue, todos]);
+
+  const addTodo = async (title: Todo['title']) => {
+    await addTodoToServer(title);
+
+    fetchTodos();
+  };
+
+  const deleteTodo = async (todoId: Todo['id']) => {
+    await deleteTodoFromServer(todoId);
+
+    fetchTodos();
+  };
+
+  const updateTodo = async (todoId: Todo['id'], completed: Todo['completed']) => {
```

Comment:
```suggestion
  const updateTodoStatus = async (todoId: Todo['id'], completed: Todo['completed']) => {
```
Так як цей метод лише міняє статус, то доречніше було б його назвати якось так.
updateTodo більше підходить для редагування самої тудушки